Shurgeken/Assets/Scripts/Gameplay/PlayerController.cs

```diff
@@ -267,19 +291,26 @@ void AirControl()
     }
 
     public void OnDamage(int dmg) {
-        data.SetAnimation(Player_Animation.DAMAGED);
+        SetAnimationWithPriority(Player_Animation.DAMAGED, 9);
         if (data.local) { DamageIndicator.DamageFlash(); }
-        busy = true;
-        busy_frames = 30;
+        being_damaged = true;
+        damage_frames = 30;
     }
 
     public void OnDie() {
-        data.SetAnimation(Player_Animation.DYING);
+        SetAnimationWithPriority(Player_Animation.DYING, 10);
         if (respawn_timer < 0)
         {
             respawn_timer = 200;
         }
     }
 
+    //bugfix: Multiple player aimations can be set on the same frame, with only one actually being sent to the animator.
+    public void SetAnimationWithPriority(Player_Animation anim, int priority) {
```

Comment:
This still seems buggy. Is there a chance that the animation will change without using this method? (I am pretty sure I saw at least one case where it does.)